explainer/Explainer.md

```diff
@@ -1,9 +1,369 @@
+confirmed (authentication) and it has been established that they have the rights to view that data
+(authorization).  In addition, data needs to be protected from interception and modification,
+and it is also necessary to guard against other risks such as tracking or inferencing of
+private data.  
+
+The WoT Discovery architecture has been defined with safeguarding security and privacy 
+as a primary goal.
+
+The WoT Architecture and WoT Thing Description documents have additional security and
+privacy considerations that are relevant to WoT Discovery.  In particular, 
+the WoT Architecture document discusses when and where secure transport such as TLS
+should be used, and when various access control mechanisms should be used.
+The problem is complicated by the fact that we often want to access IoT devices
+on a LAN where the Certificate Authority (CA) system available on the rest of the internet is not easy
+to apply.  In general we require strong controls on the distribution of data from
+WoT Things in general and WoT Discovery results in particular when access is public, and 
+best-effort controls on private networks such as LANs.
+
+#### Key Requirement: Scalability
+WoT Discovery should be usable on small devices with only a few MB of RAM to 
+large cloud services providing the ability to search thousands of IoT services
+potentially available in a city. 
+
+We did this by separating out first-contact and detailed search mechanisms,
+as described in the next section.
+The first-contact mechanisms can be chosen based on the scenario (i.e.
+broadcast mDNS-SD might be suitable on a LAN, a search engine would be suitable
+for a Smart City), while the detailed search mechanisms are RESTful web services
+simple enough to implement on individual IoT devices but with good potential for scalability for
+large cloud-based systems.
+
+## Architecture
+In general, WoT Discovery is about distribution of TDs, while
+privacy and security are about restricting distribution of information
+only to verified authorized parties.  Taken together, WoT Discovery should enable
+the distribution of TDs only to verified and authorized parties.
+
+The WoT Discovery architecture is designed to address these challenges.
+We use a two-phase approach.  In the first "first-contact" phase, which we call "Introductions",
+we support any number of open discovery mechanisms whose results, however,
+are constrained to simple URLs.  These URLs should not contain any metadata themselves.
+Instead, they have to be resolved into detailed-search "Exploration" services which can be used
+to retreive metadata. Exploration services, however, are tasked with 
+authenticating and checking the authorizations of requests, and protecting
+queries and the transmission of results, and can use best
+practices from web services to do so, including (for example) OAuth and TLS.
+Exploration services implement a scalable RESTful web API.
```

Comment:
I agree, what about saying: Exploration services might implement a scalable RESTful web API.? However, if the explainer will live and die with the current status of the spec document. It might be ok to assume Exploration like TDDs or RESTFul we API.

Reply:
Good point!   I have pushed an update tweaking this paragraph and explicitly mentioning both directories and self-description.